lib/rules/module-files-only.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @fileoverview Ensures that all files are modules (either import or export)
```

Comment:
I don't think this is really TypeScript only rules.

I'm using the the TypeScript parser because we're testing against `export =` and `import = require`. This rule could also be used in ES6+ javascript files to prevent ambiguity in the processor. [Note the rule I based it on](https://github.com/benmosher/eslint-plugin-import/blob/HEAD/docs/rules/unambiguous.md)

With that in mind, there is only one other rule that might be considered "TypeScript only", 1no-arrow-for-class-property`. And really that one may be usable in ES6+ in the future. So I don't think putting them in a separate folder makes much sense.